README.md

```diff
@@ -43,114 +43,280 @@
 
 # featurevisor-swift
 
-This repository is a work in progress to port the [Featurevisor](https://featurevisor.com) JavaScript SDK to Swift.
-
-We are not ready yet. Please come back later.
+This repository is a port of the [Featurevisor](https://featurevisor.com) JavaScript SDK to Swift.
 
 ## Installation
 
-With [SPM](https://www.swift.org/package-manager/), add the package under `dependencies` in `Package.swift` file:
+### Swift Package Manager
+
+The [Swift Package Manager](https://swift.org/package-manager/) is a tool for automating the distribution of Swift code and is integrated into the `swift` compiler.
+
+Once you have your Swift package set up, adding Featurevisor as a dependency is as easy as adding it to the `dependencies` value of your `Package.swift`.
+
+```swift
+dependencies: [
+    .package(url: "https://github.com/featurevisor/featurevisor-swift.git", .upToNextMajor(from: "X.Y.Z"))
+]
+```
+
+## Usage
+
+### Setting up Featurevisor SDK
+We would like to be able to set up the Featurevisor SDK instance once and reuse the same instance everywhere.
+
+The SDK can be initialized in two different ways depending on your needs.
+
+### Synchronous
+You can fetch the data file content on your own and just pass it via options.
+
+```swift
+import FeaturevisorSDK
+
+let datafileContent: DatafileContent = ...
+var options: InstanceOptions = .default
+options.datafile = datafileContent
+
+let featurevisor = try createInstance(options: options)
+```
+
+### Asynchronous
+If you want to delegate the responsibility of fetching the datafile to the SDK.
 
 ```swift
-// swift-tools-version:4.0
-import PackageDescription
+import FeaturevisorSDK
 
-let package = Package(
-  name: "ExampleApp",
-  dependencies: [
-    // add the repo with desired semver as a dependency here
-    .package(url: "git@github.com:featurevisor/featurevisor-swift.git", .exact("X.Y.Z"))
-  ],
-  .target(
-    name: "ExampleApp",
-    dependencies: [
-      // add the name of module from the dependency in your desired targets
-      "FeaturevisorSDK"
-    ]
-  )
-)
+var options: InstanceOptions = .default
+options.datafileUrl = "https://cdn.yoursite.com/production/datafile-tag-all.json" 
+let featurevisor = try createInstance(options: options)
 ```
 
-Building your application will fetch and install everything locally:
+If you need to take further control on how the datafile is fetched, you can pass a custom `handleDatafileFetch` function
 
+```swift
+public typealias DatafileFetchHandler = (_ datafileUrl: String) -> Result<DatafileContent, Error>
 ```
-$ swift build
+
+```swift
+import FeaturevisorSDK
+
+var options: InstanceOptions = .default
+options.handleDatafileFetch = { datafileUrl in
+    ...
```

Comment:
are we supposed to return an object following `DatafileContent` type here?

Reply:
Yep, as `handleDatafileFetch` is a following type `public typealias DatafileFetchHandler = (_ datafileUrl: String) -> Result<DatafileContent, Error>`